src/pages/Swap/LimitOrderBox/components/LimitOrderForm/LimitOrderForm.tsx

```diff
@@ -0,0 +1,508 @@
+import { Web3Provider } from '@ethersproject/providers'
+import { formatUnits, parseUnits } from '@ethersproject/units'
+import { ChainId, Currency, JSBI, Price, Token, TokenAmount } from '@swapr/sdk'
+
+import dayjs from 'dayjs'
+import dayjsUTCPlugin from 'dayjs/plugin/utc'
+import { useCallback, useEffect, useLayoutEffect, useRef, useState } from 'react'
+import { usePrevious } from 'react-use'
+import { Flex } from 'rebass'
+
+import { ButtonPrimary } from '../../../../../components/Button'
+import { AutoColumn } from '../../../../../components/Column'
+import { CurrencyInputPanel } from '../../../../../components/CurrencyInputPanel'
+import { ApprovalState, useApproveCallback } from '../../../../../hooks/useApproveCallback'
+import { useHigherUSDValue } from '../../../../../hooks/useUSDValue'
+import { useNotificationPopup } from '../../../../../state/application/hooks'
+import { useCurrencyBalances } from '../../../../../state/wallet/hooks'
+import { maxAmountSpend } from '../../../../../utils/maxAmountSpend'
+import AppBody from '../../../../AppBody'
+import { getQuote, getVaultRelayerAddress, signLimitOrder, submitLimitOrder } from '../../api'
+import { LimitOrderFormContext } from '../../contexts/LimitOrderFormContext'
+import { LimitOrderKind, OrderExpiresInUnit, SerializableLimitOrder } from '../../interfaces'
+import { getInitialState } from '../../utils'
+import { ApprovalFlow } from '../ApprovalFlow'
+import { OrderExpiryField } from '../OrderExpiryField'
+import { OrderLimitPriceField } from '../OrderLimitPriceField'
+import SwapTokens from '../SwapTokens'
+import { AutoRow, MaxAlert } from './styles'
+import { checkMaxOrderAmount, formatMaxValue } from './utils'
+
+dayjs.extend(dayjsUTCPlugin)
+
+interface HandleCurrencyAmountChangeParams {
+  currency: Currency
+  amountWei: string
+  amountFormatted: string
+  updatedLimitOrder?: SerializableLimitOrder
+}
+
+export interface LimitOrderFormProps {
+  provider: Web3Provider
+  chainId: ChainId
+  account: string
+}
+
+/**
+ * The Limit Order Form is the base component for all limit order forms.
+ */
+export function LimitOrderForm({ account, provider, chainId }: LimitOrderFormProps) {
```

Comment:
This components is too big. There is too much logic together, and coupled to the UI. It would be great to separate UI from logic. Maybe in another PR but I think this is urgent.

Reply:
This component is highly connected to the UI because they're hooks. We can't move them outside.